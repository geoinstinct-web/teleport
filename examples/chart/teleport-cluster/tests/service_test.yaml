suite: Service
templates:
  - service.yaml
tests:
  - it: uses a LoadBalancer by default
    set:
      clusterName: teleport.example.com
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Service
      - equal:
          path: spec.type
          value: LoadBalancer
      - matchSnapshot: {}

  - it: uses a ClusterIP when service.type=ClusterIP
    set:
      clusterName: teleport.example.com
      service:
        type: ClusterIP
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Service
      - equal:
          path: spec.type
          value: ClusterIP
      - matchSnapshot: {}

  - it: sets AWS annotations when chartMode=aws
    set:
      clusterName: teleport.example.com
      chartMode: aws
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Service
      - equal:
          path: spec.type
          value: LoadBalancer
      - equal:
          path: metadata.annotations.service\.beta\.kubernetes\.io/aws-load-balancer-type
          value: nlb
      - equal:
          path: metadata.annotations.service\.beta\.kubernetes\.io/aws-load-balancer-backend-protocol
          value: tcp
      - equal:
          path: metadata.annotations.service\.beta\.kubernetes\.io/aws-load-balancer-cross-zone-load-balancing-enabled
          value: "true"
      - matchSnapshot: {}

  - it: sets service annotations when specified
    values:
      - ../.lint/annotations.yaml
    asserts:
      - equal:
          path: metadata.annotations.kubernetes\.io/service
          value: test-annotation
      - equal:
          path: metadata.annotations.kubernetes\.io/service-different
          value: 5
      - matchSnapshot: {}
