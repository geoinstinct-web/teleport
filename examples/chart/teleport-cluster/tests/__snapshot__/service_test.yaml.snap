sets AWS annotations when chartMode=aws:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp
        service.beta.kubernetes.io/aws-load-balancer-cross-zone-load-balancing-enabled: "true"
        service.beta.kubernetes.io/aws-load-balancer-type: nlb
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ports:
      - name: https
        port: 443
        protocol: TCP
        targetPort: 3080
      - name: sshproxy
        port: 3023
        protocol: TCP
        targetPort: 3023
      - name: k8s
        port: 3026
        protocol: TCP
        targetPort: 3026
      - name: sshtun
        port: 3024
        protocol: TCP
        targetPort: 3024
      - name: mysql
        port: 3036
        protocol: TCP
        targetPort: 3036
      selector:
        app: RELEASE-NAME
      type: LoadBalancer
sets service annotations when specified:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      annotations:
        kubernetes.io/service: test-annotation
        kubernetes.io/service-different: 5
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ports:
      - name: https
        port: 443
        protocol: TCP
        targetPort: 3080
      - name: sshproxy
        port: 3023
        protocol: TCP
        targetPort: 3023
      - name: k8s
        port: 3026
        protocol: TCP
        targetPort: 3026
      - name: sshtun
        port: 3024
        protocol: TCP
        targetPort: 3024
      - name: mysql
        port: 3036
        protocol: TCP
        targetPort: 3036
      selector:
        app: RELEASE-NAME
      type: LoadBalancer
uses a ClusterIP when service.type=ClusterIP:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ports:
      - name: https
        port: 443
        protocol: TCP
        targetPort: 3080
      - name: sshproxy
        port: 3023
        protocol: TCP
        targetPort: 3023
      - name: k8s
        port: 3026
        protocol: TCP
        targetPort: 3026
      - name: sshtun
        port: 3024
        protocol: TCP
        targetPort: 3024
      - name: mysql
        port: 3036
        protocol: TCP
        targetPort: 3036
      selector:
        app: RELEASE-NAME
      type: ClusterIP
uses a LoadBalancer by default:
  1: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app: RELEASE-NAME
      name: RELEASE-NAME
      namespace: NAMESPACE
    spec:
      ports:
      - name: https
        port: 443
        protocol: TCP
        targetPort: 3080
      - name: sshproxy
        port: 3023
        protocol: TCP
        targetPort: 3023
      - name: k8s
        port: 3026
        protocol: TCP
        targetPort: 3026
      - name: sshtun
        port: 3024
        protocol: TCP
        targetPort: 3024
      - name: mysql
        port: 3036
        protocol: TCP
        targetPort: 3036
      selector:
        app: RELEASE-NAME
      type: LoadBalancer
